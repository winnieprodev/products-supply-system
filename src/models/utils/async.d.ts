export interface Error {
  message: string;
}

export interface Async<T> {
  isLoading: boolean;
  error: Error | undefined;
  data: T | undefined;
}
