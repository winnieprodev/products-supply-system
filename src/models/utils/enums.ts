export enum WrapperMode {
  full = "full",
  singleLine = "singleLine",
  tiny = "tiny",
}
