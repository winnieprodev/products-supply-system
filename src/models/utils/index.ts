export * from "./async.d";
export * from "./enums";
