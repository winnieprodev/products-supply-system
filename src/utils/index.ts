export * from "./routeUtils";
export * from "./supplierUtils";
export * from "./testUtils";
