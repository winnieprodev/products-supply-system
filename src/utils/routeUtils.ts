import { Empty } from "./types";

export interface Route<T> {
  route: string;
  use: (params?: T) => string;
}

export const createRoute = <T = Empty>(route: string): Route<T> => ({
  route,
  use: (params?: T) => {
    let digestedRoute = route;
    if (params !== undefined) {
      digestedRoute = Object.entries(params).reduce((acc, [key, value]) => {
        return acc.replace(new RegExp(`:${key}\\??`), value);
      }, route);
    }
    return digestedRoute.replace(/:\w+\??/, "");
  },
});
