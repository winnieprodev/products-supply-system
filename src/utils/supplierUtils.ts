import { SupplierType } from "routes";

export const generateMono = (title: string): string =>
  title
    .split(" ")
    .map((word) => word.toUpperCase())
    .filter((word) => word !== "PTE" && word !== "PTE.")
    .filter((word) => word !== "LTD")
    .map((word) => word.replace("(", "")[0])
    .join("")
    .substring(0, 3);

export const getSupplierTypeName = (type: SupplierType): string => {
  switch (type) {
    case "warehouse":
      return "Warehouse";
    case "local":
      return "Local Suppliers";
  }
};
