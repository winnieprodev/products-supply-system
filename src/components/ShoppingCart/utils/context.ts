import { Async, Cart, ProductCartInfo } from "models";
import React from "react";
import { mockAsync, mockCart } from "utils";

interface ShoppingCartContextProps {
  cartId: Async<number>;
  cart: Async<Cart> | undefined;
  setCart: (cart: Async<Cart> | undefined) => void;
  setProduct: (product: ProductCartInfo) => void;
  removeProduct: (productId: string) => void;
}

export const EMPTY_CART = { products: [] };
const mockFunction = () => {};

export const ShoppingCartContext = React.createContext<ShoppingCartContextProps>(
  {
    cartId: mockAsync(0),
    cart: mockAsync(mockCart(0)),
    setCart: mockFunction,
    setProduct: mockFunction,
    removeProduct: mockFunction,
  },
);
