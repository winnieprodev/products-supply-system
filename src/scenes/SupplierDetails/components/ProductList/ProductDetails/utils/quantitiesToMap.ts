import { Map } from "immutable";
import { CartQuantity, QuantityType } from "models";

export const quantitiesToMap = (quantities: CartQuantity[]) => {
  return Map<QuantityType, number>(
    quantities.map(({ type, amount }) => [type, amount] as const) as Iterable<
      [QuantityType, number]
    >,
  );
};
