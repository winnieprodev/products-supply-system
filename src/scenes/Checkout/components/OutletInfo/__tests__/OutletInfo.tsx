import { render } from "@testing-library/react";
import { mockOutlet } from "utils";

import OutletRadio from "../OutletInfo";

describe("OutletRadio", () => {
  describe("Snapshot tests", () => {
    test("it matches the snapshot", () => {
      const { container } = render(<OutletRadio item={mockOutlet("id")} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
