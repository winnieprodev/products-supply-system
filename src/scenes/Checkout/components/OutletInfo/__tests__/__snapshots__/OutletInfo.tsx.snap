// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutletRadio Snapshot tests it matches the snapshot 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  width: 46px;
  height: 46px;
  margin: 10px;
  margin-left: 0;
  background: #F3F3F3;
  text-align: center;
  border-radius: 10px;
  font-size: 20px;
  text-align: center;
}

.c1 > * {
  opacity: 0.2;
  padding: 12px 0;
}

.c2 {
  font-size: 18px;
  text-transform: uppercase;
  font-weight: 600;
  margin-bottom: 5px;
}

.c3 {
  font-size: 14px;
  font-weight: 600;
}

.c4 {
  font-size: 13px;
  margin: 6px 0 2px;
}

.c5 {
  font-size: 13px;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-house-user fa-w-18 "
      data-icon="house-user"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 576 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M570.69,236.27,512,184.44V48a16,16,0,0,0-16-16H432a16,16,0,0,0-16,16V99.67L314.78,10.3C308.5,4.61,296.53,0,288,0s-20.46,4.61-26.74,10.3l-256,226A18.27,18.27,0,0,0,0,248.2a18.64,18.64,0,0,0,4.09,10.71L25.5,282.7a21.14,21.14,0,0,0,12,5.3,21.67,21.67,0,0,0,10.69-4.11l15.9-14V480a32,32,0,0,0,32,32H480a32,32,0,0,0,32-32V269.88l15.91,14A21.94,21.94,0,0,0,538.63,288a20.89,20.89,0,0,0,11.87-5.31l21.41-23.81A21.64,21.64,0,0,0,576,248.19,21,21,0,0,0,570.69,236.27ZM288,176a64,64,0,1,1-64,64A64,64,0,0,1,288,176ZM400,448H176a16,16,0,0,1-16-16,96,96,0,0,1,96-96h64a96,96,0,0,1,96,96A16,16,0,0,1,400,448Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    class=""
  >
    <div
      class="c2"
    >
      STV
    </div>
    <div
      class="c3"
    >
      Star Vista
    </div>
    <div
      class="c4"
    >
      #B1-39, 1 Vista Exchange Green Singapore 138617
    </div>
    <div
      class="c5"
    >
      66944033
    </div>
  </div>
</div>
`;
